sdks/python/src/opik/api_objects/dataset/dataset.py

```diff
@@ -53,6 +54,23 @@ def insert(
             for item in items
         ]
 
+        # Remove duplicates if they already exist
+        deduplicated_items = []
+        for item in items:
+            item_hash = utils.compute_content_hash(item)
```

Comment:
Minor: you don't need to compute a hash for this, you can just go with the content payload, assuming Python equality works correct with this particular payload.. This approach is valid anyway. It's just a matter of space vs computation.